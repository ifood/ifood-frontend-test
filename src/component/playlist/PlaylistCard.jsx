/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
  Card,
  CardActionArea,
  CardActions,
  CardContent,
  CardMedia,
  Divider,
  Tooltip,
  Typography,
} from '@material-ui/core';
import { Person } from '@material-ui/icons';

const UserContainer = styled.div`
  align-items: center;
  display: flex;
  padding: 8px;
`;
const BoldText = styled.span`
  font-weight: 700;
`;

function PlaylistCard(props) {
  const { playlist } = props;

  if (playlist == null) {
    return null;
  }

  const {
    id,
    name,
    description,
    images,
    owner,
    tracks,
    external_urls,
  } = playlist;

  function handleClickCard() {
    if (external_urls?.spotify != null) {
      window.open(external_urls.spotify);
    }
  }

  /* eslint-disable-next-line react/no-danger */
  const dangerHtml = <span dangerouslySetInnerHTML={{ __html: description }} />;

  return (
    <Card key={id} onClick={handleClickCard}>
      <CardActionArea onClick={handleClickCard}>
        <CardMedia
          component="img"
          alt={name}
          image={images?.[0]?.url}
          title={name}
        />
        <CardContent>
          <Tooltip title={name} aria-label="description">
            <Typography variant="h5" noWrap>
              <BoldText>{name}</BoldText>
            </Typography>
          </Tooltip>
          <Typography variant="caption" gutterBottom>
            {`${tracks.total} tracks`}
          </Typography>
          <Tooltip title={dangerHtml} aria-label="description">
            <Typography
              variant="subtitle2"
              color="textSecondary"
              gutterBottom
              noWrap>
              {dangerHtml}
            </Typography>
          </Tooltip>
        </CardContent>
        <Divider light />
        <CardActions>
          <UserContainer>
            <Person color="secondary" style={{ paddingRight: '8px' }} />
            <Typography variant="subtitle2" color="textSecondary">
              {owner.display_name}
            </Typography>
          </UserContainer>
        </CardActions>
      </CardActionArea>
    </Card>
  );
}

PlaylistCard.propTypes = {
  playlist: PropTypes.object,
};

export default PlaylistCard;
