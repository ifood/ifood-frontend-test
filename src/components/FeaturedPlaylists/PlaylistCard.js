import React from 'react';

const PlaylistCard = (props) => (
  <div className="col-4">
    <div className="card">
    <img className="card-img-top" src={props.imageUrl} alt={props.name} />
    <div className="card-body">
      <h5 className="card-title">{props.name}</h5>
      <p className="card-text">Total tracks: {props.tracks.total}</p>
      <a href={props.externalUrl} className="btn btn-primary">See on Spotify</a>
    </div>
  </div>
  </div>
);

export default PlaylistCard;