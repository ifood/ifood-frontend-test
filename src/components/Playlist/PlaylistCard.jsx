import React from 'react';

import styled from 'styled-components';
import { randomColors, mainColor, white, black } from '../style/colors';

export const Card = styled.div`
    position: relative;
    width: 200px;
    height: 200px;
    display: flex;
    flex: 0 1 auto;
    align-content: flex-start;
    padding: 20px;
    flex-flow: column nowrap;
    justify-content: space-around;
    background: ${(props) => props.backgroundColor || mainColor};
    margin-bottom: 20px;

    div {
        flex-grow: 1;
        margin-right: 5px;

        &:last-child {
            margin-right: 0;
        }
    }

    @media (max-width: 1024px) {
        width: 100%;

        div {
            margin-right: 0;
        }
    }
`;

export const CardImage = styled.div`
    width: 80px;
    height: 80px;
    background: url(${(props) =>
        props.backgroundImage || 'https://dummyimage.com/80x80/000/fff&text=x'});
    background-repeat: no-repeat;
    background-size: 80px 80px;
    position: absolute;
    top: 20px;
    left: 20px;
    opacity: 0.6;

    @media (max-width: 1024px) {
        opacity: 1;
    }
`;

export const Name = styled.h1`
    font-family: 'Oswald', sans-serif;
    font-size: 36px;
    font-weight: 700;
    text-transform: uppercase;
    text-align: right;
    color: ${white};
`;

export const Author = styled.h1`
    font-family: 'Oswald', sans-serif;
    font-size: 16px;
    text-align: right;
    color: ${black};
`;

export const Description = styled.p`
    margin-top: 5px;
    font-family: 'Noto Sans JP', sans-serif;
    font-size: 16px;
    color: ${white};
`;

const PlayListCard = ({ name, description, owner, images }) => {
    const { display_name: displayName, href } = owner;
    const backgroundImage = images[0].url;
    const randomNumber = Math.floor(Math.random() * 10);

    return (
        <Card backgroundColor={randomColors[randomNumber]}>
            <CardImage backgroundImage={backgroundImage} />
            <Name>{name}</Name> <Author>by {displayName}</Author>
            <Description>{description}</Description>
        </Card>
    );
};

export default PlayListCard;
