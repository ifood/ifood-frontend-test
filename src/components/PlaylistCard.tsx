import { FontAwesomeIcon as Icon } from '@fortawesome/react-fontawesome';
import React, { Fragment, PureComponent } from 'react';
import Img from 'react-image';
import styled from 'styled-components';

import { IPlaylist } from '../api/spotify';

interface IProps {
  playlist: IPlaylist;
}

const CARD_SIZE = '15rem';

const StyledCardContainer = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  margin: 1.6em;
  width: ${CARD_SIZE};

  /* truncate playlist name */
  .playlist-name {
    width: ${CARD_SIZE};
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    font-size: 1.6em;
  }
`;

const StyledImageContainer = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;

  height: ${CARD_SIZE};
  width: ${CARD_SIZE};

  & img {
    max-height: ${CARD_SIZE};
    max-width: ${CARD_SIZE};
  }
`;

const DiscPlaceholder = () => (
  <Icon className='has-text-light' icon='compact-disc' size='10x' />
);

export default class PlaylistCard extends PureComponent<IProps> {
  public render() {
    const { playlist } = this.props;
    return (
      <StyledCardContainer>
        <a href={playlist.external_urls.spotify} target='_blank'>
          <StyledImageContainer>
            <Img
              className='image'
              src={playlist.images[0].url}
              loader={<DiscPlaceholder />}
              unloader={'failed to load image'}
            />
          </StyledImageContainer>
        </a>
        <div
          className='playlist-name has-text-centered has-text-weight-light'
          title={playlist.name}
        >
          {playlist.name}
        </div>
      </StyledCardContainer>
    );
  }
}
