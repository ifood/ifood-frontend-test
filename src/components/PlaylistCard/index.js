import React from "react";

export default function PlaylistCard({ playlist }) {
  const { images, collaborative, public } = playlist;
  return (
    <div
      tabindex="0"
      role="button"
      className="playlist-card"
      aria-label={`Playlist named ${playlist.name}`}
    >
      <div className="playlist-card__dark-cover" />
      <img className="playlist-card__img" src={images[0].url}></img>
    </div>
  );
}
