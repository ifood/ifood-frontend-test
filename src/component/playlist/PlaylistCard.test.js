/* eslint-disable camelcase */
/* eslint-disable no-unused-expressions */
/* eslint-disable no-undef, react/jsx-filename-extension, react/prop-types */
import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Card } from '@material-ui/core';
import PlaylistCard from './PlaylistCard';

const playlist = {
  description: 'Description test',
  external_urls: {
    spotify: 'https://href-test',
  },
  href: 'https://hrf-test',
  id: '1HJ38ZSXZ123SA',
  images: [
    {
      url: 'https://img-test',
    },
  ],
  name: 'PlaylistTest',
  owner: {
    display_name: 'Spotify',
    external_urls: {
      spotify: 'https://open.spotify.com/user/spotify',
    },
    href: 'https://api.spotify.com/v1/users/spotify',
    id: 'spotify',
  },
  tracks: {
    total: 62,
  },
  type: 'playlist',
};

describe('<PlaylistCard />', () => {
  it('snapshot', async () => {
    const wrapper = mount(<PlaylistCard playlist={playlist} />);
    jestExpect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should return null when playlist prop not provided', () => {
    const card = shallow(<PlaylistCard />);
    expect(card.type()).to.be.null;
  });

  describe('whend card clicked', () => {
    global.open = jest.fn();
    const card = mount(<PlaylistCard />);

    beforeEach(() => {
      global.open.mockClear();
    });

    it('should open a new window with external_url', () => {
      card.setProps({ playlist });
      card.find(Card).simulate('click');

      expect(global.open.mock.calls.length).to.be.equal(1);

      expect(global.open.mock.calls[0][0]).to.be.equal(
        playlist.external_urls.spotify,
      );
    });

    it('should not open a new window when no external_url specified', () => {
      const { external_urls, ...rest } = playlist;
      card.setProps({ playlist: rest });
      card.find(Card).simulate('click');

      expect(global.open.mock.calls.length).to.be.equal(0);
    });
  });
});
