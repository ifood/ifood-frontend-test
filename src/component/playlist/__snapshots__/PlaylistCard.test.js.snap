// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaylistCard /> snapshot 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px;
}

.c0 {
  font-weight: 700;
}

<PlaylistCard
  playlist={
    Object {
      "description": "Description test",
      "external_urls": Object {
        "spotify": "https://href-test",
      },
      "href": "https://hrf-test",
      "id": "1HJ38ZSXZ123SA",
      "images": Array [
        Object {
          "url": "https://img-test",
        },
      ],
      "name": "PlaylistTest",
      "owner": Object {
        "display_name": "Spotify",
        "external_urls": Object {
          "spotify": "https://open.spotify.com/user/spotify",
        },
        "href": "https://api.spotify.com/v1/users/spotify",
        "id": "spotify",
      },
      "tracks": Object {
        "total": 62,
      },
      "type": "playlist",
    }
  }
>
  <WithStyles(ForwardRef(Card))
    key="1HJ38ZSXZ123SA"
    onClick={[Function]}
  >
    <ForwardRef(Card)
      classes={
        Object {
          "root": "MuiCard-root",
        }
      }
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(Paper))
        className="MuiCard-root"
        elevation={1}
        onClick={[Function]}
      >
        <ForwardRef(Paper)
          className="MuiCard-root"
          classes={
            Object {
              "elevation0": "MuiPaper-elevation0",
              "elevation1": "MuiPaper-elevation1",
              "elevation10": "MuiPaper-elevation10",
              "elevation11": "MuiPaper-elevation11",
              "elevation12": "MuiPaper-elevation12",
              "elevation13": "MuiPaper-elevation13",
              "elevation14": "MuiPaper-elevation14",
              "elevation15": "MuiPaper-elevation15",
              "elevation16": "MuiPaper-elevation16",
              "elevation17": "MuiPaper-elevation17",
              "elevation18": "MuiPaper-elevation18",
              "elevation19": "MuiPaper-elevation19",
              "elevation2": "MuiPaper-elevation2",
              "elevation20": "MuiPaper-elevation20",
              "elevation21": "MuiPaper-elevation21",
              "elevation22": "MuiPaper-elevation22",
              "elevation23": "MuiPaper-elevation23",
              "elevation24": "MuiPaper-elevation24",
              "elevation3": "MuiPaper-elevation3",
              "elevation4": "MuiPaper-elevation4",
              "elevation5": "MuiPaper-elevation5",
              "elevation6": "MuiPaper-elevation6",
              "elevation7": "MuiPaper-elevation7",
              "elevation8": "MuiPaper-elevation8",
              "elevation9": "MuiPaper-elevation9",
              "outlined": "MuiPaper-outlined",
              "root": "MuiPaper-root",
              "rounded": "MuiPaper-rounded",
            }
          }
          elevation={1}
          onClick={[Function]}
        >
          <div
            className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(CardActionArea))
              onClick={[Function]}
            >
              <ForwardRef(CardActionArea)
                classes={
                  Object {
                    "focusHighlight": "MuiCardActionArea-focusHighlight",
                    "focusVisible": "Mui-focusVisible",
                    "root": "MuiCardActionArea-root",
                  }
                }
                onClick={[Function]}
              >
                <WithStyles(ForwardRef(ButtonBase))
                  className="MuiCardActionArea-root"
                  focusVisibleClassName="Mui-focusVisible"
                  onClick={[Function]}
                >
                  <ForwardRef(ButtonBase)
                    className="MuiCardActionArea-root"
                    classes={
                      Object {
                        "disabled": "Mui-disabled",
                        "focusVisible": "Mui-focusVisible",
                        "root": "MuiButtonBase-root",
                      }
                    }
                    focusVisibleClassName="Mui-focusVisible"
                    onClick={[Function]}
                  >
                    <button
                      className="MuiButtonBase-root MuiCardActionArea-root"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <WithStyles(ForwardRef(CardMedia))
                        alt="PlaylistTest"
                        component="img"
                        image="https://img-test"
                        title="PlaylistTest"
                      >
                        <ForwardRef(CardMedia)
                          alt="PlaylistTest"
                          classes={
                            Object {
                              "img": "MuiCardMedia-img",
                              "media": "MuiCardMedia-media",
                              "root": "MuiCardMedia-root",
                            }
                          }
                          component="img"
                          image="https://img-test"
                          title="PlaylistTest"
                        >
                          <img
                            alt="PlaylistTest"
                            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
                            src="https://img-test"
                            title="PlaylistTest"
                          />
                        </ForwardRef(CardMedia)>
                      </WithStyles(ForwardRef(CardMedia))>
                      <WithStyles(ForwardRef(CardContent))>
                        <ForwardRef(CardContent)
                          classes={
                            Object {
                              "root": "MuiCardContent-root",
                            }
                          }
                        >
                          <div
                            className="MuiCardContent-root"
                          >
                            <WithStyles(ForwardRef(Tooltip))
                              aria-label="description"
                              title="PlaylistTest"
                            >
                              <ForwardRef(Tooltip)
                                aria-label="description"
                                classes={
                                  Object {
                                    "arrow": "MuiTooltip-arrow",
                                    "popper": "MuiTooltip-popper",
                                    "popperArrow": "MuiTooltip-popperArrow",
                                    "popperInteractive": "MuiTooltip-popperInteractive",
                                    "tooltip": "MuiTooltip-tooltip",
                                    "tooltipArrow": "MuiTooltip-tooltipArrow",
                                    "tooltipPlacementBottom": "MuiTooltip-tooltipPlacementBottom",
                                    "tooltipPlacementLeft": "MuiTooltip-tooltipPlacementLeft",
                                    "tooltipPlacementRight": "MuiTooltip-tooltipPlacementRight",
                                    "tooltipPlacementTop": "MuiTooltip-tooltipPlacementTop",
                                    "touch": "MuiTooltip-touch",
                                  }
                                }
                                title="PlaylistTest"
                              >
                                <WithStyles(ForwardRef(Typography))
                                  aria-describedby={null}
                                  aria-label="description"
                                  className=""
                                  noWrap={true}
                                  onBlur={[Function]}
                                  onFocus={[Function]}
                                  onMouseLeave={[Function]}
                                  onMouseOver={[Function]}
                                  onTouchEnd={[Function]}
                                  onTouchStart={[Function]}
                                  title="PlaylistTest"
                                  variant="h5"
                                >
                                  <ForwardRef(Typography)
                                    aria-describedby={null}
                                    aria-label="description"
                                    className=""
                                    classes={
                                      Object {
                                        "alignCenter": "MuiTypography-alignCenter",
                                        "alignJustify": "MuiTypography-alignJustify",
                                        "alignLeft": "MuiTypography-alignLeft",
                                        "alignRight": "MuiTypography-alignRight",
                                        "body1": "MuiTypography-body1",
                                        "body2": "MuiTypography-body2",
                                        "button": "MuiTypography-button",
                                        "caption": "MuiTypography-caption",
                                        "colorError": "MuiTypography-colorError",
                                        "colorInherit": "MuiTypography-colorInherit",
                                        "colorPrimary": "MuiTypography-colorPrimary",
                                        "colorSecondary": "MuiTypography-colorSecondary",
                                        "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                        "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                        "displayBlock": "MuiTypography-displayBlock",
                                        "displayInline": "MuiTypography-displayInline",
                                        "gutterBottom": "MuiTypography-gutterBottom",
                                        "h1": "MuiTypography-h1",
                                        "h2": "MuiTypography-h2",
                                        "h3": "MuiTypography-h3",
                                        "h4": "MuiTypography-h4",
                                        "h5": "MuiTypography-h5",
                                        "h6": "MuiTypography-h6",
                                        "noWrap": "MuiTypography-noWrap",
                                        "overline": "MuiTypography-overline",
                                        "paragraph": "MuiTypography-paragraph",
                                        "root": "MuiTypography-root",
                                        "srOnly": "MuiTypography-srOnly",
                                        "subtitle1": "MuiTypography-subtitle1",
                                        "subtitle2": "MuiTypography-subtitle2",
                                      }
                                    }
                                    noWrap={true}
                                    onBlur={[Function]}
                                    onFocus={[Function]}
                                    onMouseLeave={[Function]}
                                    onMouseOver={[Function]}
                                    onTouchEnd={[Function]}
                                    onTouchStart={[Function]}
                                    title="PlaylistTest"
                                    variant="h5"
                                  >
                                    <h5
                                      aria-describedby={null}
                                      aria-label="description"
                                      className="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap"
                                      onBlur={[Function]}
                                      onFocus={[Function]}
                                      onMouseLeave={[Function]}
                                      onMouseOver={[Function]}
                                      onTouchEnd={[Function]}
                                      onTouchStart={[Function]}
                                      title="PlaylistTest"
                                    >
                                      <styled.span>
                                        <span
                                          className="c0"
                                        >
                                          PlaylistTest
                                        </span>
                                      </styled.span>
                                    </h5>
                                  </ForwardRef(Typography)>
                                </WithStyles(ForwardRef(Typography))>
                                <ForwardRef(Popper)
                                  anchorEl={
                                    .c0 {
  font-weight: 700;
}

<h5
                                      aria-label="description"
                                      class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap"
                                      title="PlaylistTest"
                                    >
                                      <span
                                        class="c0"
                                      >
                                        PlaylistTest
                                      </span>
                                    </h5>
                                  }
                                  className="MuiTooltip-popper"
                                  id={null}
                                  open={false}
                                  placement="bottom"
                                  popperOptions={
                                    Object {
                                      "modifiers": Object {
                                        "arrow": Object {
                                          "element": null,
                                          "enabled": false,
                                        },
                                      },
                                    }
                                  }
                                  transition={true}
                                />
                              </ForwardRef(Tooltip)>
                            </WithStyles(ForwardRef(Tooltip))>
                            <WithStyles(ForwardRef(Typography))
                              gutterBottom={true}
                              variant="caption"
                            >
                              <ForwardRef(Typography)
                                classes={
                                  Object {
                                    "alignCenter": "MuiTypography-alignCenter",
                                    "alignJustify": "MuiTypography-alignJustify",
                                    "alignLeft": "MuiTypography-alignLeft",
                                    "alignRight": "MuiTypography-alignRight",
                                    "body1": "MuiTypography-body1",
                                    "body2": "MuiTypography-body2",
                                    "button": "MuiTypography-button",
                                    "caption": "MuiTypography-caption",
                                    "colorError": "MuiTypography-colorError",
                                    "colorInherit": "MuiTypography-colorInherit",
                                    "colorPrimary": "MuiTypography-colorPrimary",
                                    "colorSecondary": "MuiTypography-colorSecondary",
                                    "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                    "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                    "displayBlock": "MuiTypography-displayBlock",
                                    "displayInline": "MuiTypography-displayInline",
                                    "gutterBottom": "MuiTypography-gutterBottom",
                                    "h1": "MuiTypography-h1",
                                    "h2": "MuiTypography-h2",
                                    "h3": "MuiTypography-h3",
                                    "h4": "MuiTypography-h4",
                                    "h5": "MuiTypography-h5",
                                    "h6": "MuiTypography-h6",
                                    "noWrap": "MuiTypography-noWrap",
                                    "overline": "MuiTypography-overline",
                                    "paragraph": "MuiTypography-paragraph",
                                    "root": "MuiTypography-root",
                                    "srOnly": "MuiTypography-srOnly",
                                    "subtitle1": "MuiTypography-subtitle1",
                                    "subtitle2": "MuiTypography-subtitle2",
                                  }
                                }
                                gutterBottom={true}
                                variant="caption"
                              >
                                <span
                                  className="MuiTypography-root MuiTypography-caption MuiTypography-gutterBottom"
                                >
                                  62 tracks
                                </span>
                              </ForwardRef(Typography)>
                            </WithStyles(ForwardRef(Typography))>
                            <WithStyles(ForwardRef(Tooltip))
                              aria-label="description"
                              title={
                                <span
                                  dangerouslySetInnerHTML={
                                    Object {
                                      "__html": "Description test",
                                    }
                                  }
                                />
                              }
                            >
                              <ForwardRef(Tooltip)
                                aria-label="description"
                                classes={
                                  Object {
                                    "arrow": "MuiTooltip-arrow",
                                    "popper": "MuiTooltip-popper",
                                    "popperArrow": "MuiTooltip-popperArrow",
                                    "popperInteractive": "MuiTooltip-popperInteractive",
                                    "tooltip": "MuiTooltip-tooltip",
                                    "tooltipArrow": "MuiTooltip-tooltipArrow",
                                    "tooltipPlacementBottom": "MuiTooltip-tooltipPlacementBottom",
                                    "tooltipPlacementLeft": "MuiTooltip-tooltipPlacementLeft",
                                    "tooltipPlacementRight": "MuiTooltip-tooltipPlacementRight",
                                    "tooltipPlacementTop": "MuiTooltip-tooltipPlacementTop",
                                    "touch": "MuiTooltip-touch",
                                  }
                                }
                                title={
                                  <span
                                    dangerouslySetInnerHTML={
                                      Object {
                                        "__html": "Description test",
                                      }
                                    }
                                  />
                                }
                              >
                                <WithStyles(ForwardRef(Typography))
                                  aria-describedby={null}
                                  aria-label="description"
                                  className=""
                                  color="textSecondary"
                                  gutterBottom={true}
                                  noWrap={true}
                                  onBlur={[Function]}
                                  onFocus={[Function]}
                                  onMouseLeave={[Function]}
                                  onMouseOver={[Function]}
                                  onTouchEnd={[Function]}
                                  onTouchStart={[Function]}
                                  title={null}
                                  variant="subtitle2"
                                >
                                  <ForwardRef(Typography)
                                    aria-describedby={null}
                                    aria-label="description"
                                    className=""
                                    classes={
                                      Object {
                                        "alignCenter": "MuiTypography-alignCenter",
                                        "alignJustify": "MuiTypography-alignJustify",
                                        "alignLeft": "MuiTypography-alignLeft",
                                        "alignRight": "MuiTypography-alignRight",
                                        "body1": "MuiTypography-body1",
                                        "body2": "MuiTypography-body2",
                                        "button": "MuiTypography-button",
                                        "caption": "MuiTypography-caption",
                                        "colorError": "MuiTypography-colorError",
                                        "colorInherit": "MuiTypography-colorInherit",
                                        "colorPrimary": "MuiTypography-colorPrimary",
                                        "colorSecondary": "MuiTypography-colorSecondary",
                                        "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                        "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                        "displayBlock": "MuiTypography-displayBlock",
                                        "displayInline": "MuiTypography-displayInline",
                                        "gutterBottom": "MuiTypography-gutterBottom",
                                        "h1": "MuiTypography-h1",
                                        "h2": "MuiTypography-h2",
                                        "h3": "MuiTypography-h3",
                                        "h4": "MuiTypography-h4",
                                        "h5": "MuiTypography-h5",
                                        "h6": "MuiTypography-h6",
                                        "noWrap": "MuiTypography-noWrap",
                                        "overline": "MuiTypography-overline",
                                        "paragraph": "MuiTypography-paragraph",
                                        "root": "MuiTypography-root",
                                        "srOnly": "MuiTypography-srOnly",
                                        "subtitle1": "MuiTypography-subtitle1",
                                        "subtitle2": "MuiTypography-subtitle2",
                                      }
                                    }
                                    color="textSecondary"
                                    gutterBottom={true}
                                    noWrap={true}
                                    onBlur={[Function]}
                                    onFocus={[Function]}
                                    onMouseLeave={[Function]}
                                    onMouseOver={[Function]}
                                    onTouchEnd={[Function]}
                                    onTouchStart={[Function]}
                                    title={null}
                                    variant="subtitle2"
                                  >
                                    <h6
                                      aria-describedby={null}
                                      aria-label="description"
                                      className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary MuiTypography-noWrap MuiTypography-gutterBottom"
                                      onBlur={[Function]}
                                      onFocus={[Function]}
                                      onMouseLeave={[Function]}
                                      onMouseOver={[Function]}
                                      onTouchEnd={[Function]}
                                      onTouchStart={[Function]}
                                      title={null}
                                    >
                                      <span
                                        dangerouslySetInnerHTML={
                                          Object {
                                            "__html": "Description test",
                                          }
                                        }
                                      />
                                    </h6>
                                  </ForwardRef(Typography)>
                                </WithStyles(ForwardRef(Typography))>
                                <ForwardRef(Popper)
                                  anchorEl={
                                    <h6
                                      aria-label="description"
                                      class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary MuiTypography-noWrap MuiTypography-gutterBottom"
                                    >
                                      <span>
                                        Description test
                                      </span>
                                    </h6>
                                  }
                                  className="MuiTooltip-popper"
                                  id={null}
                                  open={false}
                                  placement="bottom"
                                  popperOptions={
                                    Object {
                                      "modifiers": Object {
                                        "arrow": Object {
                                          "element": null,
                                          "enabled": false,
                                        },
                                      },
                                    }
                                  }
                                  transition={true}
                                />
                              </ForwardRef(Tooltip)>
                            </WithStyles(ForwardRef(Tooltip))>
                          </div>
                        </ForwardRef(CardContent)>
                      </WithStyles(ForwardRef(CardContent))>
                      <WithStyles(ForwardRef(Divider))
                        light={true}
                      >
                        <ForwardRef(Divider)
                          classes={
                            Object {
                              "absolute": "MuiDivider-absolute",
                              "flexItem": "MuiDivider-flexItem",
                              "inset": "MuiDivider-inset",
                              "light": "MuiDivider-light",
                              "middle": "MuiDivider-middle",
                              "root": "MuiDivider-root",
                              "vertical": "MuiDivider-vertical",
                            }
                          }
                          light={true}
                        >
                          <hr
                            className="MuiDivider-root MuiDivider-light"
                          />
                        </ForwardRef(Divider)>
                      </WithStyles(ForwardRef(Divider))>
                      <WithStyles(ForwardRef(CardActions))>
                        <ForwardRef(CardActions)
                          classes={
                            Object {
                              "root": "MuiCardActions-root",
                              "spacing": "MuiCardActions-spacing",
                            }
                          }
                        >
                          <div
                            className="MuiCardActions-root MuiCardActions-spacing"
                          >
                            <styled.div>
                              <div
                                className="c1"
                              >
                                <ForwardRef
                                  color="secondary"
                                  style={
                                    Object {
                                      "paddingRight": "8px",
                                    }
                                  }
                                >
                                  <WithStyles(ForwardRef(SvgIcon))
                                    color="secondary"
                                    style={
                                      Object {
                                        "paddingRight": "8px",
                                      }
                                    }
                                  >
                                    <ForwardRef(SvgIcon)
                                      classes={
                                        Object {
                                          "colorAction": "MuiSvgIcon-colorAction",
                                          "colorDisabled": "MuiSvgIcon-colorDisabled",
                                          "colorError": "MuiSvgIcon-colorError",
                                          "colorPrimary": "MuiSvgIcon-colorPrimary",
                                          "colorSecondary": "MuiSvgIcon-colorSecondary",
                                          "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                          "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                          "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                          "root": "MuiSvgIcon-root",
                                        }
                                      }
                                      color="secondary"
                                      style={
                                        Object {
                                          "paddingRight": "8px",
                                        }
                                      }
                                    >
                                      <svg
                                        aria-hidden={true}
                                        className="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                                        focusable="false"
                                        style={
                                          Object {
                                            "paddingRight": "8px",
                                          }
                                        }
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                                        />
                                      </svg>
                                    </ForwardRef(SvgIcon)>
                                  </WithStyles(ForwardRef(SvgIcon))>
                                </ForwardRef>
                                <WithStyles(ForwardRef(Typography))
                                  color="textSecondary"
                                  variant="subtitle2"
                                >
                                  <ForwardRef(Typography)
                                    classes={
                                      Object {
                                        "alignCenter": "MuiTypography-alignCenter",
                                        "alignJustify": "MuiTypography-alignJustify",
                                        "alignLeft": "MuiTypography-alignLeft",
                                        "alignRight": "MuiTypography-alignRight",
                                        "body1": "MuiTypography-body1",
                                        "body2": "MuiTypography-body2",
                                        "button": "MuiTypography-button",
                                        "caption": "MuiTypography-caption",
                                        "colorError": "MuiTypography-colorError",
                                        "colorInherit": "MuiTypography-colorInherit",
                                        "colorPrimary": "MuiTypography-colorPrimary",
                                        "colorSecondary": "MuiTypography-colorSecondary",
                                        "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                        "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                        "displayBlock": "MuiTypography-displayBlock",
                                        "displayInline": "MuiTypography-displayInline",
                                        "gutterBottom": "MuiTypography-gutterBottom",
                                        "h1": "MuiTypography-h1",
                                        "h2": "MuiTypography-h2",
                                        "h3": "MuiTypography-h3",
                                        "h4": "MuiTypography-h4",
                                        "h5": "MuiTypography-h5",
                                        "h6": "MuiTypography-h6",
                                        "noWrap": "MuiTypography-noWrap",
                                        "overline": "MuiTypography-overline",
                                        "paragraph": "MuiTypography-paragraph",
                                        "root": "MuiTypography-root",
                                        "srOnly": "MuiTypography-srOnly",
                                        "subtitle1": "MuiTypography-subtitle1",
                                        "subtitle2": "MuiTypography-subtitle2",
                                      }
                                    }
                                    color="textSecondary"
                                    variant="subtitle2"
                                  >
                                    <h6
                                      className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
                                    >
                                      Spotify
                                    </h6>
                                  </ForwardRef(Typography)>
                                </WithStyles(ForwardRef(Typography))>
                              </div>
                            </styled.div>
                          </div>
                        </ForwardRef(CardActions)>
                      </WithStyles(ForwardRef(CardActions))>
                      <span
                        className="MuiCardActionArea-focusHighlight"
                      />
                      <WithStyles(memo)
                        center={false}
                      >
                        <ForwardRef(TouchRipple)
                          center={false}
                          classes={
                            Object {
                              "child": "MuiTouchRipple-child",
                              "childLeaving": "MuiTouchRipple-childLeaving",
                              "childPulsate": "MuiTouchRipple-childPulsate",
                              "ripple": "MuiTouchRipple-ripple",
                              "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                              "rippleVisible": "MuiTouchRipple-rippleVisible",
                              "root": "MuiTouchRipple-root",
                            }
                          }
                        >
                          <span
                            className="MuiTouchRipple-root"
                          >
                            <TransitionGroup
                              childFactory={[Function]}
                              component={null}
                              exit={true}
                            />
                          </span>
                        </ForwardRef(TouchRipple)>
                      </WithStyles(memo)>
                    </button>
                  </ForwardRef(ButtonBase)>
                </WithStyles(ForwardRef(ButtonBase))>
              </ForwardRef(CardActionArea)>
            </WithStyles(ForwardRef(CardActionArea))>
          </div>
        </ForwardRef(Paper)>
      </WithStyles(ForwardRef(Paper))>
    </ForwardRef(Card)>
  </WithStyles(ForwardRef(Card))>
</PlaylistCard>
`;
